fmanimashaun/budget-app
# Project completion

## Description:
In this PR, we have implemented a new splash screen for our application. The splash screen features a logo and two buttons for user authentication - "Sign In" and "Sign Up". Users can navigate to the respective sign-in or sign-up pages by clicking these buttons.

## Changes Made:

### Home page (categories page):

- When the user logs in, they are presented with the categories page.
- For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
- When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- There is a button "add a new category" at the bottom that brings the user to the page to create a new category.


### Transactions page:

- For a given category, the list of transactions is presented, ordered by the most recent.
- At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

### Add a `new category` page:

- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

### Add a `new transaction` page

- The user fills out a form to create a new transaction with:
  - name (mandatory)
  - amount (mandatory)
  - categories (mandatory at least one)
- The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
- When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

## How To Test:

1. Run the unit tests for the controller and the view.
2. Run the system test to ensure the buttons work correctly.

## screenshots

![Screen Shot 2024-01-19 at 16 33 14](https://github.com/fmanimashaun/budget-app/assets/22054464/21f99af0-4528-447e-8c65-60a11b32644f)
